elastic/kibana
# [Canvas] Handle element changing into a filter

Fixes #33605

If an expression is edited to become a filter, or a different type of filter, then the filter value on that element would become incorrect.  

This fixes that issue by making sure the filter value on the element is of the correct type. 

**Steps to reproduce/test:**

1. Add an element with the expression `filters | render as=debug`, so you can see all the filter values
2. Add a datatable element to your workpad
3. Change the expression to `timefilterControl compact=true column=@timestamp | render`
4. Note that the filter value in the debug element does not change, or is invalid

Also try changing a dropdownFilter expression to a timeFilter expression and vice versa and ensure the correct filter value ends up on the element and shows in the debug.
